lib/typed_struct.ex

```diff
@@ -116,24 +175,80 @@ defmodule TypedStruct do
     end
   end
 
+  @doc false
+  def __typedrecord__(name, block, opts) do
+    tag = Keyword.get(opts, :tag, name)
+
+    quote do
+      Enum.each(unquote(@accumulating_attrs), fn attr ->
+        Module.register_attribute(__MODULE__, attr, accumulate: true)
+      end)
+
```

Comment:
This was intentional, because quite frankly I couldn't think of a good reason for supporting plugins in records.  We can add this feature later if desired.